File: problems_on_please/CpyAug2012/contests/day06/numbers/solutions/solution.cpp
#include<iostream>
#include <cstdio>
using namespace std;

void Gen(int n, int k, string Prefix)
{
	if(n==0)
		cout<<Prefix<<endl;
	else
	{
		for(char c='0';c<'0'+k;++c)
		Gen(n-1,k,Prefix+string(1,c)+" ");
	}
}

int main()
{
    freopen("numbers.in", "r", stdin);
    freopen("numbers.out", "w", stdout);
	int n, k;
	cin>>n>>k;
	Gen(n,k,"");
	return 0;
}

File: problems_on_please/CpyAug2012/contests/day06/numbers/solutions/solution.py
#!/usr/bin/python3

fin = open('numbers.in', 'r')
fout = open('numbers.out', 'w')

def gen(n, k, prefix):
    if n == 0:
        print(" ".join(map(str, prefix)), file=fout)
        return
    for i in range(k):
        gen(n - 1, k, prefix + [i])

n, k = map(int, fin.read().strip().split())
gen(n, k, [])

